internal/livelint/check_can_visit_public_app.go

```diff
@@ -1,16 +1,49 @@
 package livelint
 
-func (n *Livelint) checkCanVisitPublicApp() CheckResult {
-	yes := n.askUserYesOrNo("The app should be working. Can you visit it from the public internet?")
+import (
+	"context"
+	"fmt"
+	"log"
+	"net/http"
+	"strconv"
 
-	if !yes {
+	apiv1 "k8s.io/api/core/v1"
+)
+
+func (n *Livelint) checkCanVisitPublicApp(namespace string, services []apiv1.Service) CheckResult {
+	ingresses, err := n.getIngressesFromServices(namespace, services)
+	if err != nil {
 		return CheckResult{
 			HasFailed:    true,
 			Message:      "You cannot visit the app from the public internet",
-			Instructions: "The issue is likely to be with the infrastructure and how the cluster is exposed",
+			Instructions: "No matching ingresses were found",
 		}
 	}
+	for _, ingress := range ingresses {
+		urls := getUrlsFromIngress(ingress)
+		for _, url := range urls {
+			req, err := http.NewRequestWithContext(context.Background(), http.MethodGet, url.String(), nil)
+			if err != nil {
+				log.Fatal(fmt.Errorf("error when creating http request for url %s: %w", url.String(), err))
+			}
+			resp, err := http.DefaultClient.Do(req)
+			if err != nil {
+				return CheckResult{
+					HasFailed: true,
+					Message:   fmt.Sprintf("You cannot visit %s from public internet: %s", url.String(), err.Error()),
+				}
+			}
+			defer resp.Body.Close()
+			if resp.StatusCode < 200 || resp.StatusCode >= 300 {
```

Comment:
Should we allow `3XX` and `4XX` codes here? I think that's valid because maybe the user just needs to be authenticated, but that would have nothing to do with not being able to reach the app. I would make this `resp.StatusCode < 200 || resp.StatusCode >= 500` but add a `Details` in the positive check result saying what the returned status code was.

Reply:
Done.